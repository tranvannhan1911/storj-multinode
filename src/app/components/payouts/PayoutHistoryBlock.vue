// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <info-block>
        <div slot="body" class="payouts-history-block">
            <h3 class="payouts-history-block__title">Payout History</h3>
            <v-button class="payouts-history-block__button" label="Download" :is-white="true" width="100%" />
        </div>
    </info-block>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import InfoBlock from '@/app/components/common/InfoBlock.vue';
import VButton from '@/app/components/common/VButton.vue';

// @vue/component
@Component({
    components: {
        InfoBlock,
        VButton,
    },
})
export default class PayoutHistoryBlock extends Vue {}
</script>

<style scoped lang="scss">
    .payouts-history-block {
        box-sizing: border-box;
        width: 100%;
        display: flex;
        flex-direction: column;
        align-items: flex-start;

        &__title {
            font-family: 'font_bold', sans-serif;
            font-size: 24px;
            color: var(--c-title);
            margin-bottom: 30px;
        }

        &__button {
            background-color: var(--c-button-gray);
            border: none !important;
        }
    }
</style>
