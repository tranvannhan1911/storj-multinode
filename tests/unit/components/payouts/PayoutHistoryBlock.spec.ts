// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

import { shallowMount } from '@vue/test-utils';

import PayoutHistoryBlock from '@/app/components/payouts/PayoutHistoryBlock.vue';

describe('PayoutHistoryBlock', (): void => {
    it('renders correctly', (): void => {
        const wrapper = shallowMount(PayoutHistoryBlock);

        expect(wrapper).toMatchSnapshot();
    });
});
