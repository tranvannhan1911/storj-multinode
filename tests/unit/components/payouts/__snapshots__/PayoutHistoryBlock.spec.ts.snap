// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayoutHistoryBlock renders correctly 1`] = `
<info-block-stub>
  <div class="payouts-history-block">
    <h3 class="payouts-history-block__title">Payout History</h3>
    <v-button-stub label="Download" width="100%" height="48px" iswhite="true" onpress="[Function]" class="payouts-history-block__button"></v-button-stub>
  </div>
</info-block-stub>
`;
